### PR TITLE
Feature 38/optimize infra

### PR DESCRIPTION
**Pull request by patricksimonian**
updated the infra code to be more production like

- added cpu resources/limits 
- added memory resrouces/limits
- utilized variables so that multiple instances of the app can be built and deployed
- added playbooks to make it easier to build and deploy

### CODE REVIEW

File: openshift/ansible/build-react.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        register: pull_request
+
+      - fail: 
+          msg: PR {{ PR }} does not exist or is closed
+        when: "'closed' in pull_request.json.state"
+      
+      - name: create build tools name space using head/pulls/{{ PR }} as ref 
+        shell: >
+          oc process -f "../templates/react-frontend/bc.yaml" --param-file="../templates/react-frontend/build.param" -p GIT_REF="pull/{{ PR }}/head" -p  IMAGE_TAG="pr-{{ PR }}" | 
+          oc apply -n "{{ tools_namespace }}" -f -
```
Comment (ShellyXueHan) [inline openshift/ansible/build-react.yaml:17]: empty line

File: openshift/ansible/build-react.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+          return_content: yes
+        register: pull_request
+
+      - fail: 
+          msg: PR {{ PR }} does not exist or is closed
+        when: "'closed' in pull_request.json.state"
+      
+      - name: create build tools name space using head/pulls/{{ PR }} as ref 
+        shell: >
+          oc process -f "../templates/react-frontend/bc.yaml" --param-file="../templates/react-frontend/build.param" -p GIT_REF="pull/{{ PR }}/head" -p  IMAGE_TAG="pr-{{ PR }}" |
```
Comment (ShellyXueHan) [inline openshift/ansible/build-react.yaml:16]: not sure if this is ideal way passing in params from both file and separate ones. Maybe consider turning the build.param file into a template and source it here?

Comment (patricksimonian) [reply openshift/ansible/build-react.yaml:16]: what do you mean about that?

Comment (ShellyXueHan) [reply openshift/ansible/build-react.yaml:16]: for example, turn the param file into a jinja template, and make those params (i.e. IMAGE_TAG) optional base on condition. that way you don't need to explicitly set them here.

Comment (ShellyXueHan) [reply openshift/ansible/build-react.yaml:16]: it's good for now, and I think you can proceed with it for sure. templating param values is just nice to have.

**Pull request merged**